src/Cli/Microsoft.DotNet.Cli.Utils/TelemetryEventEntry.cs

```diff
@@ -47,6 +48,28 @@ void Handler(object sender, InstrumentationEventArgs eventArgs)
         }
     }
 
+    public sealed class PerformanceMeasurement : IDisposable
+    {
+        private readonly Stopwatch _timer;
+        private readonly Dictionary<string, double> _data;
+        private readonly string _name;
+
+        public PerformanceMeasurement(Dictionary<string, double> data, string name)
+        {
+            // Measurement is a no-op if we don't have a dictionary to store the entry.
+            if (data == null)
+            {
+                return;
+            }
+
+            _data = data;
+            _name = name;
+            _timer = Stopwatch.StartNew();
+        }
+
+        public void Dispose() => _data?.Add(_name, _timer.Elapsed.TotalMilliseconds);
```

Comment:
Question, does the runtime trigger dispose immediately at the end of the using block or does it do it lazily? In other words, will this change end up adding (likely very small amounts of) time to the perf tracking because we're relying on an implicit dispose rather than an explicit elapsed time recording as it was before.

Reply:
That was the exact question I asked both Jacques and Smit. It will immediately run `Dispose` when exiting the scope of the `using`. So, statement order is:
- Last statement inside the `using`
- Dispose
- First statement outside the `using`

It is deterministic. We used this pattern heavily in our C# code generator on a previous team. You can see an [example here](https://github.com/Azure/autorest.csharp/blob/0ee0276021228fb9a03ed359cd8a621fefd36bb0/src/AutoRest.CSharp/LowLevel/Generation/LowLevelClientWriter.cs#L259-L272) and throughout that file (and many of the files for that project).